linode/helper/instance.go

```diff
@@ -22,18 +22,46 @@ const (
 var bootEvents = []linodego.EventAction{linodego.ActionLinodeBoot, linodego.ActionLinodeReboot}
 
 // set bootConfig = 0 if using existing boot config
-func RebootInstance(ctx context.Context, d *schema.ResourceData, entityID int,
+func RebootInstance(ctx context.Context, d *schema.ResourceData, linodeID int,
 	meta interface{}, bootConfig int,
 ) diag.Diagnostics {
+	client := meta.(*ProviderMeta).Client
+	ctx, cancel := context.WithTimeout(
+		ctx,
+		time.Duration(GetDeadlineSeconds(ctx, d))*time.Second,
+	)
+	defer cancel()
+	return diag.FromErr(rebootInstance(ctx, linodeID, &client, bootConfig))
+}
+
+func FrameworkRebootInstance(
+	linodeID int,
```

Comment:
Could this function also accept a context that is passed into `context.WithTimeout(...)` so log context, etc. will be inherited?

Reply:
Good point! I think inheriting the timeout from the context and removing the timeout argument might be the better pattern here, although that's definitely up for discussion